django/chat/responses.py

```diff
@@ -417,6 +417,15 @@ async def add_files_to_library():
         input = response_message.parent.text
         source_nodes = retriever.retrieve(input)
 
+        if chat.options.qa_source_order == "reading_order":
+            source_nodes = sorted(
+                source_nodes,
+                key=lambda x: (
+                    x.node.relationships["1"].node_id,
+                    x.metadata["chunk_number"],
```

Comment:
Trying to remember the sorting code in the Gradio app...
I would have thought that first, it would sort by document relevance (i.e. average or max score of the nodes for each document), then by chunk number.
In any case, I think that is a better way to sort them.

Reply:
Docs are now ordered by max relevance score in each one.
Made a note to try average at some point for comparison, but for now max is easier to implement and fairly intuitive to me.